detect_secrets/core/secrets_collection.py

```diff
@@ -180,7 +187,11 @@ def trim(
                     # Only update line numbers if we're tracking them.
                     existing_secret.line_number = secret.line_number
```

Comment:
Looking at your implementation, it looks like it could be fairly trivial to track the line number of each of the occurrences found and report them as it was mentioned in #493.